utils/docker-compose/README.md

```diff
@@ -0,0 +1,227 @@
+`$PEER_CMD` must also be set to the location of binary or script that will start
+ the peer.  **Docker version 17.06.2-ce or higher is needed**
+
+## Steps
+1. Download the necessary fabric binaries. Run the
+[bootstrap script](scripts/bootstrap.sh) which will download the Fabric 1.4.3
+into a local bin directory. If you already have the binaries downloaded in your
+path, this step can be skipped. **Fabric 1.4.3 versions of configtxgen and
+cryptogen are required to use the configurations above.**
+```
+./scripts/bootstrap.sh
+```
+1. Build the peer image in `utils/docker/peer` directory which is defined by the
+peer [Dockerfile](../docker/peer/Dockerfile). This step
+assumes you have already built the [fabric-private-chaincode base image](../docker/base/Dockerfile).
+Take a look at building the docker dev environment in the main [README](../../README.md#docker).
+After you have created the base image, run the following to create a modified
+peer image and the plugins necessary to start the peer.  `$FPC_PATH` is the
+location fabric-private-chaincode repository on your host machine.
+```
+cd $FPC_PATH/utils/docker/peer
+docker build -t hyperledger/fabric-peer-fpc .
+```
+By default the image will clone the master branch on
+https://github.com/hyperledger-labs/fabric-private-chaincode. If you want to use
+a different fork of the repo or a different branch you provide
+FPC_REPO_URL and FPC_REPO_BRANCH as build args.
+```
+cd $FPC_PATH/utils/docker/peer
+docker build -t hyperledger/fabric-peer-fpc --build-arg FPC_REPO_URL=<repo-url> --build-arg FPC_REPO_BRANCH=<repo-branch> .
+```
+If you want to build the peer image using your local copy of your repo you can
+use the same build args, but specify `file:///tmp/build-src/.git` as the
+`FPC_REPO_URL`. You will also need to create the image at the root of this repo
+so that the local repo will be in the build context for the docker daemon.
+```
+cd $FPC_PATH
+docker build -t hyperledger/fabric-peer-fpc -f utils/docker/peer/Dockerfile --build-arg FPC_REPO_URL=file:///tmp/build-src/.git --build-arg FPC_REPO_BRANCH=$(git rev-parse --abbrev-ref HEAD) .
+```
+2. Generate the cryptographic material needed for the network by running the
+[generate](scripts/generate.sh) script. Cryptogen will be used to generate all the
+credentials needed based on the configuration filesabove and place them in the
+`network-config/crypto-config` directory.  Configtxgen will be used to create
+the genesis block which is used to start up the orderer as well as the peer
+create channel configuration transaction. These will be placed in the
+`network-config/config` directory. The `crypto-config` & `config` directory will
+be mounted into every container of the FPC network as specified in the
+docker-compose file. **This script is not
+idempotent and will delete the contents of `crypto-config` & `config` when run
+to ensure a clean start.**
+```
+cd $FPC_PATH/utils/docker-compose
+scripts/generate.sh
```

Comment:
when I run this step, I get the following error:
```
  docker-compose git:(pr/188) ✗ ./scripts/generate.sh
./scripts/generate.sh: line 18: cryptogen: command not found
Failed to generate crypto material...
```
I indeed did step 1 downloading the binaries and therefore did not expect this error here. Double checking that download step fetched cryptogen shows it is there but not recognised by the `generate.sh` script.

```
bin git:(pr/188) ✗ pwd
/Users/marcusbrandenburger/Developer/gocode/src/github.com/hyperledger-labs/fabric-private-chaincode/utils/docker-compose/scripts/bin
➜  bin git:(pr/188) ✗ ll
total 288408
-rwxr-xr-x  1 marcusbrandenburger  staff    16M Aug 26 23:34 configtxgen
-rwxr-xr-x  1 marcusbrandenburger  staff    17M Aug 26 23:34 configtxlator
-rwxr-xr-x  1 marcusbrandenburger  staff    11M Aug 26 23:34 cryptogen
-rwxr-xr-x  1 marcusbrandenburger  staff    16M Aug 26 23:34 discover
-rwxr-xr-x  1 marcusbrandenburger  staff    14M Aug 26 23:31 fabric-ca-client
-rwxr-xr-x  1 marcusbrandenburger  staff   9.4M Aug 26 23:34 idemixgen
-rwxr-xr-x  1 marcusbrandenburger  staff    24M Aug 26 23:34 orderer
-rwxr-xr-x  1 marcusbrandenburger  staff    29M Aug 26 23:34 peer
```

Reply:
when I change `generate.sh` it works
```
-export PATH=${SCRIPT_DIR}/../bin:${PWD}:$PATH
+export PATH=${SCRIPT_DIR}/bin:${PWD}:$PATH
```

Reply:
interestingly, this did work for me (although i've invoked it as `bash ./scripts/generate.sh` as the script identified itself wrongly as bourne shell at that time). Looking at the script now it looks correct and what you suggest wrong.  Do you have somehow managed to get the bin directory inside the scripts dir?

Reply:
Oh, interesting, where did you run the bootstrap script from? The expected location is supposed to be at the same level as the `SCRIPT_DIR`. Let me try to fix the bootstrap script so it doesn't download relative to where the script is run but relative to the script location

Reply:
Okay, so the script is rather large and would require a bit of refactoring to be able to download relative to the script location for this problem. I will try to address that in a followup pr. For now I put more info on the download step so users are aware they have to run the script in a specific location.

Reply:
actually, it is very easy to fix (2 lines): will come in my upcoming PR :-)